web/src/message_viewport.js

```diff
@@ -59,11 +59,11 @@ export function message_viewport_info() {
     const $element_just_below_us = $("#compose");
 
     res.visible_top =
-        $element_just_above_us.offset().top + $element_just_above_us.safeOuterHeight();
+        $element_just_above_us.offset().top + $element_just_above_us.outerHeight() ?? 0;
```

Comment:
This change is wrong (along with most of the other similar changes) because `??` has lower operator precedence than `+`.

Reply:
@andersk put it in a bracket?

Reply:
@andersk take a look.

Btw for expressions using `+=`, RHS with nullish coalescing operator has higher precendence so didn't change that. If you want I could change that too.